### PR TITLE
LASB-1960 - integration test for AWS SQS

### PR DESCRIPTION
**Pull request by lkm0287**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-1960)

Integration test for SQS.

Basically works with a few caveats:

- Requires a localstack docker image in the local registry.

- A couple of errors in the logs when running the test around SQS credentials (not really sure why) and shutting down the queue that could be looked in at a future date (the queues are actually shut down but looks like it could be done more gracefully).

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Comment by mtac50** [comment]
Nice work @lkm0287! It looks like local stack integrates nicely with CircleCI if we want to use it with any of the other services so @alexdesi could use it without any issues in CDA (assuming there is a Ruby Gem for it...). As for running it in AWS, I think this should also be possible as the base image we use for the pipeline has docker installed. We could put a copy of the local stack image in our AWS container registry and then pull it down before we run the gradle build command in the build.sh script.  We'd have to test it but I think it should work!

File: maat-court-data-api/src/test/java/gov/uk/courtdata/link/service/CreateLinkListenerEndToEndTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+        sqsTemplate.convertAndSend(queueName, messageJson);
+
+        await().atMost(3, SECONDS).untilAsserted(() -> {
+            // assert the SQS is in expected state
+            assertThat(numberOfMessagesInLinkQueue()).isEqualTo(0);
+            assertThat(numberOfMessagesNotVisibleInLinkQueue()).isEqualTo(0);
+
+            // verify behaviour as expected
+            verify(createLinkService, times(ONCE)).saveAndLink(any());
```
Comment (skasthuri36326) [inline maat-court-data-api/src/test/java/gov/uk/courtdata/link/service/CreateLinkListenerEndToEndTest.java:68]: Nice work! Is it possible to configure DLQ and test the exception scenarios?

Comment (lkm0287) [reply maat-court-data-api/src/test/java/gov/uk/courtdata/link/service/CreateLinkListenerEndToEndTest.java:68]: Thanks, I believe you can set a DWQ: https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/examples-sqs-dead-letter-queues.html and test it. You also can test the exception scenarios.

**Review by mtac50 — APPROVED** [review]

**Review by mikereiddigital — APPROVED** [review]
lg2m

**Pull request merged**